### PR TITLE
Fix bug in MatrixFree::get_face_iterator

### PR DESCRIPTION
**Pull request by peterrum**
this PR will also add a missing test...

### CODE REVIEW

**Comment by peterrum** [comment]
pring @kronbichler

File: tests/matrix_free/iterators.cc [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+
+// tests matrix-free get_cell_iterator and ger_face_iterator for
```
Comment (kronbichler) [inline tests/matrix_free/iterators.cc]:
```suggestion
// tests matrix-free get_cell_iterator and get_face_iterator for
```

Comment (peterrum) [reply tests/matrix_free/iterators.cc]: Thanks for the review! Fixed.

**Comment by bangerth** [comment]
One of the automated checks fails:
```
matrix_free/iterators.debug: DIFF failed. ------ First 20 lines of numdiff output:

----------------

##8       #:5   <== 11

##8       #:5   ==> 20

@ Absolute error = 9.0000000000e+0, Relative error = 8.1818181818e-1

----------------

##9       #:5   <== 12

##9       #:5   ==> 21

@ Absolute error = 9.0000000000e+0, Relative error = 7.5000000000e-1

----------------

##10      #:5   <== 13

##10      #:5   ==> 22

@ Absolute error = 9.0000000000e+0, Relative error = 6.9230769231e-1

----------------

##11      #:5   <== 20

##11      #:5   ==> 23

@ Absolute error = 3.0000000000e+0, Relative error = 1.5000000000e-1

----------------

##12      #:5   <== 21

##12      #:5   ==> 30

@ Absolute error = 9.0000000000e+0, Relative error = 4.2857142857e-1
```

**Comment by bangerth** [comment]
Still fails.

**Pull request merged**